### PR TITLE
Support `std::unique_ptr`.

### PR DESCRIPTION
**Pull request by zoecarver**
Currently the synthesized members consume but that's OK for now. At least it's correct.

### CODE REVIEW

**Comment by zoecarver** [comment]
@swift-ci please test

File: test/Interop/Cxx/stdlib/std-unique-ptr.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-experimental-cxx-interop)
+//
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import StdUniquePtr
+#if os(Linux)
+import CxxStdlib
+// FIXME: import CxxStdlib.string once libstdc++ is split into submodules.
```
Comment (hyp) [inline test/Interop/Cxx/stdlib/std-unique-ptr.swift:9]: That's not going to happen. Can you just import CxxStdlib?

Comment (egorzhdan) [reply test/Interop/Cxx/stdlib/std-unique-ptr.swift:9]: We have a bunch of existing tests that do this, I'll update them to just import CxxStdlib.

**Comment by zoecarver** [comment]
@swift-ci please test windows

**Comment by zoecarver** [comment]
Shall we guard move only types on libc++ for the time being or try to fix the libstdc++ issue?

**Comment by zoecarver** [comment]
@swift-ci please test

**Comment by zoecarver** [comment]
@swift-ci please smoke test

**Pull request closed without merging**